client/dashboard/sites/overview/index.tsx

```diff
@@ -158,12 +208,24 @@ function SiteOverview( {
 					spacing={ spacing }
 					alignment="flex-start"
 				>
-					<LatestActivityCard site={ site } isCompact={ isSmallViewport } />
-					{ ! isSelfHostedJetpackConnectedSite && ! site.is_wpcom_staging_site && (
-						<VStack spacing={ spacing } justify="start">
-							<DomainsCard site={ site } isCompact={ isSmallViewport } />
-							<DIFMUpsellCard site={ site } />
-						</VStack>
+					{ ! isLoadingSiteDomains && (
+						<>
+							<LatestActivityCard site={ site } isCompact={ isSmallViewport } />
+							{ shouldShowSecondaryUpsellStack && (
+								<VStack spacing={ spacing } justify="start">
+									{ ( () => {
+										if ( shouldShowDomainTransferUpsellCard ) {
+											return <DomainTransferUpsellCard />;
+										}
+										if ( shouldShowDomainUpsellCard ) {
+											return <DomainUpsellCard site={ site } />;
+										}
+										return null;
+									} )() }
```

Comment:
I'm wondering if this would be the more expected (and so more quickly groked) way to write. Despite the fact there's the nasty double mention of the same boolean.

```suggestion
									{ shouldShowDomainTransferUpsellCard && <DomainTransferUpsellCard /> }
									{ shouldShowDomainUpsellCard && ! shouldShowDomainTransferUpsellCard && <DomainUpsellCard site={ site } /> }
```

Reply:
I wonder if `shouldShowDomainUpsellCard && ! shouldShowDomainTransferUpsellCard` should go in useSecondaryUpsellStack instead.

That is:
`const  shouldShowDomainUpsellCard = ! shouldShowDomainTransferUpsellCard && getShouldShowDomainUpsellCard( domains )`

Reply:
On further thought, I'll move it into the function so that all the abstraction logic is in the same place.